engine/src/io/DataLoader.cpp

```diff
@@ -215,8 +214,7 @@ std::unique_ptr<ral::frame::BlazingTable> data_loader::load_data(
 
 void data_loader::get_schema(Schema & schema, std::vector<std::pair<std::string, cudf::type_id>> non_file_columns) {
 	std::vector<std::shared_ptr<arrow::io::RandomAccessFile>> files;
-	bool firstIteration = true;
-	std::vector<data_handle> handles = this->provider->get_all();
+	std::vector<data_handle> handles = this->provider->get_some(5); // we only need one to get the schema, but we will get a couple in-case the first couple files are empty
```

Comment:
This is really not correct. Don't get some and assume its fine. get one by one until one is not empty.

Reply:
Good point, i will fix this